src/rtgsfit.c

```diff
@@ -71,11 +71,14 @@ void make_basis(
             basis[i_grid + N_GRID] = 0.0;
         }        
     }
+    
     // could use gradient from previous iteration.  should apply mask 
-    gradient_z(flux_norm, &basis[2*N_GRID]);   
+    if (N_PLS == 3)
+    {
+        gradient_z(flux_norm, &basis[2*N_GRID]);   
+    }
```

Comment:
What should happen if N_PLS is not 3?

Reply:
currently only works if N_PLS is 2 or 3 probably should make it more explicit and have both options